onnx_xla/operator_registry.cc

```diff
@@ -0,0 +1,51 @@
+#include "onnx_xla/operator_registry.h"
+#include <iostream>
+namespace onnx_xla  {
+  
+  OperatorRegistry::OperatorRegisterOnce::OperatorRegisterOnce(const Symbol& nodeKind, TranslationFunction translator)  {
+    auto& map = OperatorRegistry::map();
+    if (map.find(nodeKind) == map.end())  {
+      map[nodeKind] = translator;
+    } else {
+      throw std::runtime_error("Operator registry error");
+    }
+  }
+
+  onnxStatus OperatorRegistry::executeTranslation(Node& n, XlaBuilder& builder, ValueOpMap& valueToOp)  {
+    auto& map = OperatorRegistry::map();
+    if (map.find(n.kind()) != map.end())  {
+      return map[n.kind()](n, builder, valueToOp);
+    } else {
+      return ONNXIFI_STATUS_UNSUPPORTED_OPERATOR;
+    }
+  }
+
+  OperatorRegistry* OperatorRegistry::registry()  {
+    static OperatorRegistry registry_;
+    return &registry_;
+  }
+
+  TranslationMap& OperatorRegistry::map()  {
+    static TranslationMap map;
+    return map;
+  }
+
+  onnxStatus translateRelu(Node& n, XlaBuilder& builder, ValueOpMap& valueToOp)  {
+    auto input = valueToOp[n.inputs()[0]];
+    auto shape = builder.GetShape(input);
+    if (!shape.ok())  {
+      throw std::runtime_error("Internal error: Unexpected operation shape");
```

Comment:
this doesn't look to be an internal error to me. in what situations will it fail to get the shape?

Reply:
I considered some of the ONNXIFI shape status codes, but those refer to input/output shapes specifically. Here we are reading in the shape of an XlaOp we created and populated. It errors if it the XlaOp is invalid for the builder as found here (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/xla_client/xla_builder.cc#L2254), all of which seem to me to be internal errors in making the graph? We could say its a model problem with ONNXIFI_STATUS_INVALID_MODEL, but it isn't clear to me this would be the reason.